### DISCUSSION TITLE
docker self hosted storage exited with code 139 when using s3 backend

### DISCUSSION THREAD

**Discussion by Anj-interactivated** [supabase/supabase#15599] [Questions] [answered]
The whole setup runs fine on a laptop but when running on a server I get this:

```supabase-imgproxy               | INFO    [2023-07-06T14:49:39Z] Completed in 107.761µs /health  request_id=SQaajL0LOu73vkp_y4GJd method=GET status=200 client_ip=127.0.0.1
supabase-storage                | running migrations
supabase-storage                | (node:1) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
supabase-storage                | 
supabase-storage                | Please migrate your code to use AWS SDK for JavaScript (v3).
supabase-storage                | For more information, check the migration guide at https://a.co/7PzMCcy
supabase-storage                | (Use `node --trace-warnings ...` to show where the warning was created)
supabase-storage                | finished migrations
supabase-storage exited with code 139
```

Anyone experienced something like this ? when running `docker compose up` with default and customised docker-compose.yml start storage service fine with

`STORAGE_BACKEND: file`

problem appears when changing it to 

```
STORAGE_BACKEND: s3

....rest of the s3 setup is provided and validated to work (on laptop)
```

Hosting is on hetzner  with ubuntu Ubuntu 22.04.2 LTS, docker24.0.2, build cb74dfc, has the issue 

tested with same exact setup of OS and docker ver on laptop runs fine , laptop with Mac os also fine

**Comment by Anj-interactivated** [comment, marked as answer]
this has been sloved by updating the aws libraries to the following 
```
    "@aws-sdk/client-s3": "3.369.0",
    "@aws-sdk/lib-storage": "3.369.0",
    "@aws-sdk/node-http-handler": "3.369.0",
    "@aws-sdk/s3-request-presigner": "3.369.0",
```
Looks like it was related to the processor architecture 
```
lscpu
Architecture:            aarch64
  CPU op-mode(s):        32-bit, 64-bit
  Byte Order:            Little Endian
CPU(s):                  8
  On-line CPU(s) list:   0-7
Vendor ID:               ARM
  Model name:            Neoverse-N1
    Model:               1
    Thread(s) per core:  1
    Core(s) per cluster: 8
    Socket(s):           -
    Cluster(s):          1
    Stepping:            r3p1
    BogoMIPS:            50.00
    Flags:               fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
...
```

Then I had to build storage image and use it in the docker-compose.yaml

Would be good to update the libs in the official repo

**Reply by dazaun** [reply]
Hey! I ran into the same issue as we also use hetzner arm64 servers. I tried to change the versions as there is no new patch yet but i can't make it work. Would you be willing to share what exactly you've changed please? That would help a lot! Thank you very much for any help that you can give me 🙏🏼

**Reply by Anj-interactivated** [reply]
1. get supabase storage repo https://github.com/supabase/storage-api 
2. update aws libraries in https://github.com/supabase/storage-api/blob/master/package.json (version below for me solved the problem but probably anything higher will be ok as well)
```
    "@aws-sdk/client-s3": "3.369.0",
    "@aws-sdk/lib-storage": "3.369.0",
    "@aws-sdk/node-http-handler": "3.369.0",
    "@aws-sdk/s3-request-presigner": "3.369.0",
```
3. build an image form the storage dockerfile
4. Edit  `docker-compose.yml` use the new image in supabase docker compose instead of original storage 
```
  storage:
    container_name: supabase-storage
    # image: supabase/storage-api:v0.41.2
    image: storage-api-test:1
```

**Reply by dazaun** [reply]
Thanks a lot! I did exactly that but when trying to build the image (i'm using docker buildx to build it for arm64) I get an error that the pakcage.json and package-lock.json should be in sync. So i updated the package-lock.json as best as i could but then i get an even weirder error with the s3 libraries. Did you have the same problems?

**Reply by Anj-interactivated** [reply]
hmm can you just delete the package-lock.json ? Cant remember any problems with building

**Reply by dazaun** [reply]
No unfortunately that doesn‘t work as npm ci requires an existing package-lock file. Did you generate the package-lock file separately or you just used the one provided in the repo?

**Reply by Anj-interactivated** [reply]
`npm install` should generate it

**Reply by dazaun** [reply]
I've tried it now by generating the package-lock but i get the same issues as before all concerning the typescript compilation `tsc -p tsconfig.json`; for example: `src/storage/backend/s3.ts(85,36): error TS2339: Property 'send' does not exist on type 'S3Client'.` I already tried to look at the files to see if there is way to fix it but i guess there is a skill issue on my side. You probably didn't run into that issue but do you maybe have an idea on how to fix it?

**Reply by Anj-interactivated** [reply]
Nah sorry. Maybe try with fresh project and check if it works first
1. `git clone https://github.com/supabase/storage-api.git`
2. `cd storage-api` and `npm i`
3. `npm run dev`

If it runs proceed to updating the aws

1. update package.json
```
    "@aws-sdk/client-s3": "3.369.0",
    "@aws-sdk/lib-storage": "3.369.0",
    "@aws-sdk/node-http-handler": "3.369.0",
    "@aws-sdk/s3-request-presigner": "3.369.0",
```
2. `npm i`
3. `npm run dev`


then create a docker image

**Reply by dazaun** [reply]
Thank you! For some reason it worked now. Maybe something was off in a file and i didn‘t notice but recloning it made it work🙏🏼
